Observer/AdyenHppDataAssignObserver.php

```diff
@@ -57,9 +75,10 @@ public function execute(Observer $observer)
     {
         // Get request fields
         $data = $this->readDataArgument($observer);
-
         $additionalData = $this->getValidatedAdditionalData($data);
-
+        $this->adyenHelper->adyenLogger->addAdyenDebug("AdyenHppDataAssignObserver--AdditionalData" . json_encode($additionalData));
+        $additionalData[self::STATE_DATA] = $this->checkoutStateDataValidator->getValidatedAdditionalData($additionalData[self::STATE_DATA]);
```

Comment:
Have you tested it with an HPP payment method which doesn't send the $additionalData[self::STATE_DATA] from the frontend? I believe it will throw a php error if it's missing.